flocker/acceptance/test_api.py

```diff
@@ -340,14 +339,16 @@ class DatasetAPITests(TestCase):
     """
     Tests for the dataset API.
     """
-    def test_dataset_creation(self):
+    @require_cluster(1)
+    def test_dataset_creation(self, cluster):
         """
         A dataset can be created on a specific node.
         """
         return create_dataset(self)
```

Comment:
`create_dataset` calls `get_test_cluster` too.  Maybe the dataset creation part of `create_dataset` should be factored into a function that takes a cluster as an argument and then all of the tests that have a `cluster` thanks to `require_cluster` can just pass it in to that new function.